### DISCUSSION TITLE
AttributeError: 'NoneType' object has no attribute 'encode'

### DISCUSSION THREAD

**Discussion by NobodyLivesHere** [earthlab-education/Earth-Analytics-2023-01-Intro#420] [Q&A04 Raster Data] [answered]
I still am not able to download the data that I need to complete the assignment from the EarthData website.  When putting in the API request through Python, I never receive a prompt to input my login info for EarthData as shown in the video.  I brought this up during the class discussion and it seemed the workaround was to just use a different browser.  I have tried in Chrome, Brave Browser, Firefox, and even locally through the desktop version of Visual Studio Code.   I am no longer getting the keyring error (but still am not prompted for my login) but now get this strange error as if there is a problem with the Appeears library itself

![image](https://github.com/earthlab-education/Earth-Analytics-2023-01-Intro/assets/88915377/33213add-093f-4043-87b0-58f9556ec37a)

I have restarted the kernels countless times, have deleted all the code and readded it.  Is there maybe some alternate way of getting the API to load the data in?

Edit: if I comment out the piece of code checking for the download folder to start the download, the error goes away but the download still never starts and still doesn't prompt for any sort of login 

Edit: dug deeper,  Seems something on my computer is preventing the getpass method within the AppearsDownloader class from executing properly.   I'm also struggling to get it to recognize my EarthData login if I add code to put it into the keyring as well, I'm not sure if there's some import issue with keyring / requests module within earthpy or something else amiss

**Comment by ellenalamont17** [comment]
You probably already checked this, but do you have pop-up blocker turned off? Mine was initially on and i was having similar issues. It worked for me when I approved the pop-up.

**Reply by NobodyLivesHere** [reply]
I have tried this to no avail.  Might have to run the PC in barebones afemode to get past it.  What's mostly confusing is why it suddenly changed from a keyring error to the Nonetype error, as the latter implies some issue with the module itself not having attributes set correctly but there shouldn't have been anything that changed in that regard

**Comment by NobodyLivesHere** [comment, marked as answer]
Figured it out.  For some reason the keyring module in the codespace never loaded in correctly so there was a mismatch when the Appeears module ran the keyring check to confirm the status of the download.  I remade the codespace and copied the code over and turned off the ad blocker settings on the browser and the prompt finally worked.  Hopefully the rest goes smoothly from here.

**Comment by eculler** [comment]
Just some context - I changed the environment so that it did not require a keyring because I couldn't get the keyring working on Codespaces. So if you were using the old environment, that is why you got the KeyRingError.

As for the NoneType error, I would need more info to debug. That error is not being thrown in my code, and the Traceback is cutting off the actual location of the error (I'm guessing `requests`?). But it sounds like it is working with the ad blocker off.

**Reply by NobodyLivesHere** [reply]
It was coming from the requests module when I expanded the error.  I think it was because I tried to use the keyring alt to get around the initial keyring error I was getting and the imports were getting confused.  I just remade the codespace to clear out any module conflicts and after that was able to get it to work with ad blocker turned off.
